src/System.Net.Http.WinHttpHandler/src/System/Net/Http/WinInetProxyHelper.cs

```diff
@@ -178,5 +181,44 @@ public bool ManualSettingsOnly
 
             return useProxy;
         }
+
+        public bool AutoDetectScriptSuccess()
+        {
+            // Only check WinHttpDetectAutoProxyConfigUrl if auto detect is set.
+            if (!AutoDetect) return false;
+
+            bool doAutoDetect = false;
+            object executionLock = new object();
+            long currentTimeStamp = DateTime.Now.Ticks;
+
+            lock (executionLock)
+            {
+                doAutoDetect = _previousTimeStampTicks == DateTime.MinValue.Ticks ?
+                    true : (currentTimeStamp - _previousTimeStampTicks > TimerPeriod ? true : false);
```

Comment:
There's no reason to special case _previousTimeStampTicks.  currentTimeStamp - DateTime.MinValue.Ticks is always going to be > TimerPeriod.